### PR TITLE
PMM-7250 Return NotFound on Get when cluster is deleted, FailedPrecondition when it's not ready, Rename Get -> GetCredentials

### PR DESCRIPTION
**Pull request by jprukner**
https://jira.percona.com/browse/PMM-7250

Right now we call Get on every cluster. What it actually returns is credentials to connect to the db cluster. That is why we decided to rename this API endpoint to GetCredentials.
As part of changes on frontend side we should call it only on demand, thus avoiding errors when user does not have sufficient permissions to view credentials.
We want to avoid returning credentials on changing cluster so this PR covers returning FailedPrecondition in this case.
We also want to return NotFound when cluster is deleting or just non-existent.

### CODE REVIEW

File: packages.dot [hunk 1]
```diff
@@ -1,5 +1,6 @@
 digraph packages {
 	"/service/cluster" -> "/service/k8sclient";
+	"/service/cluster" -> "/service/k8sclient/kubectl";
```
Comment (AlekSi) [inline packages.dot:3]: May we avoid that dependency on the implementation details?

Comment (AlekSi) [reply packages.dot:3]: See https://github.com/percona-platform/dbaas-controller/pull/137

Comment (jprukner) [reply packages.dot]: :+1: I removed the dependency.

File: service/k8sclient/k8sclient_test.go [hunk 2]
```diff
@@ -49,6 +49,16 @@ func TestK8sClient(t *testing.T) {
 
 	l := logger.Get(ctx)
 
+	t.Run("Get non-existing clusters", func(t *testing.T) {
+		t.Parallel()
+		if _, err := client.GetPSMDBCluster(ctx, "d0ca1166b638c-psmdb"); !errors.Is(err, ErrNotFound) {
```
Comment (BupycHuk) [inline service/k8sclient/k8sclient_test.go:54]: let's use `assert.EqualError`

Comment (BupycHuk) [reply service/k8sclient/k8sclient_test.go:54]: here and in other places

File: service/k8sclient/k8sclient.go [hunk 3]
```diff
@@ -522,13 +522,23 @@ func (c *K8sClient) DeleteXtraDBCluster(ctx context.Context, name string) error
 	return nil
 }
 
+// ErrNotFound should be returned when referenced resource does not exist
+// inside Kubernetes cluster.
+var ErrNotFound error = errors.New("resource was not found in Kubernetes cluster")
```
Comment (BupycHuk) [inline service/k8sclient/k8sclient.go:527]: let's move this to other errors on the top

File: service/k8sclient/k8sclient.go [hunk 4]
```diff
@@ -522,13 +522,23 @@ func (c *K8sClient) DeleteXtraDBCluster(ctx context.Context, name string) error
...
+
 // GetXtraDBCluster returns an XtraDB cluster credentials.
 func (c *K8sClient) GetXtraDBCluster(ctx context.Context, name string) (*XtraDBCredentials, error) {
 	var cluster pxc.PerconaXtraDBCluster
 	err := c.kubeCtl.Get(ctx, string(perconaXtraDBClusterKind), name, &cluster)
 	if err != nil {
+		if errors.Is(err, kubectl.ErrNotFound) {
+			return nil, errors.Wrap(ErrNotFound, "cannot get XtraDb cluster")
```
Comment (BupycHuk) [inline service/k8sclient/k8sclient.go:535]: `cannot get XtraDB cluster credentials`

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: service/k8sclient/k8sclient_test.go (hunk 2 above)
Comment (jprukner) [reply service/k8sclient/k8sclient_test.go]: done

File: service/k8sclient/k8sclient.go (hunk 3 above)
Comment (jprukner) [reply service/k8sclient/k8sclient.go:527]: done

File: service/k8sclient/k8sclient.go (hunk 4 above)
Comment (jprukner) [reply service/k8sclient/k8sclient.go:535]: I created const error template, used both for xtradb and mongo.

**Review by AlekSi — CHANGES_REQUESTED** [review]
CI fails hard

**Comment by jprukner** [comment]
> CI fails hard

I downgraded minikube version. Tests are working now.

File: service/k8sclient/k8sclient.go [hunk 5]
```diff
@@ -551,7 +565,6 @@ func (c *K8sClient) GetXtraDBCluster(ctx context.Context, name string) (*XtraDBC
 	return credentials, nil
 }
 
-// GetXtraDBCluster returns an XtraDB cluster credentials.
```
Comment (BupycHuk) [inline service/k8sclient/k8sclient.go:554]: 👍🏼

**Review by BupycHuk — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]
Looks good to me. 👍

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -86,6 +86,9 @@ jobs:
       - name: Configure minikube
         run: |
           sudo apt install -y conntrack
+          sudo curl -Lo /tmp/minikube https://github.com/kubernetes/minikube/releases/download/v1.16.0/minikube-linux-amd64
+          sudo chmod +x /tmp/minikube
+          sudo mv /tmp/minikube $(which minikube)
```
Comment (AlekSi) [inline .github/workflows/ci.yml:91]: Please switch to using https://github.com/percona-platform/actions-setup-minikube and adding minikube version to `strategy.matrix`

**Review by AlekSi — APPROVED** [review]
LGTM with a suggestion. Please do not re-request review from me for this PR :)

File: .github/workflows/ci.yml (hunk 6 above)
Comment (AlekSi) [reply .github/workflows/ci.yml:91]: (if that's easy to do in this PR)

Comment (jprukner) [reply .github/workflows/ci.yml:91]: I will do this in separate PR.

**Review by artemgavrilov — APPROVED** [review]

**Comment by jprukner** [comment]
rebased on pmm/2.15

**Pull request merged**